onnxruntime/python/tools/transformers/gpt2_beamsearch_helper.py

```diff
@@ -130,30 +133,217 @@ def forward(
             current_step_scores.view(self.config.batch_size * self.config.beam_size, -1),
         )
 
+class GPT2LMHeadModel_BeamSearchStepConfiguration(GPT2LMHeadModel):
+    """Here we wrap a class for Onnx model conversion for GPT2LMHeadModel with past state and one 
+    step beam search with configuration support."""
+
+    def __init__(self, 
+                 config, 
+                 batch_size, 
+                 beam_size, 
+                 ignore_eos=False,
+                 temperature=1.0, 
+                 repetition_penalty=1.0, 
+                 excluded_token_ids=None, 
+                 length_penalty=1.0, 
+                 do_sample=False, 
+                 do_sample_top_p=1, 
+                 do_sample_top_k=0):
+        super().__init__(config)
+        self.config.batch_size = batch_size
+        self.config.beam_size = beam_size
+        self.config.ignore_eos = ignore_eos
+        self.config.temperature = temperature
+        self.config.repetition_penalty = repetition_penalty
+        self.config.excluded_token_ids = excluded_token_ids
+        self.config.length_penalty = length_penalty
+        self.config.do_sample = do_sample
+        self.config.do_sample_top_p = do_sample_top_p
+        self.config.do_sample_top_k = do_sample_top_k
+    
+    @staticmethod
+    def collapse_first_two_dims(tensor):
+        return tensor.view(-1, *tensor.size()[2:])
+    
+    @staticmethod
+    def top_k_top_p_filtering(log_probs, top_p=1.0, top_k=0):
+        '''Set tail event (out of top_p) to a big negative number'''
+        sorted_log_probs, sorted_indices = torch.sort(log_probs, descending=True)
+        cumulative_probs = torch.cumsum(sorted_log_probs.exp(), dim=-1)
```

Comment:
Add a comment that CumSum for float16 need onnx opset version 14.

Reply:
I have tested on Pytorch and ORT 1.7.0 and it works fine. Could you clarify?